minitwit/src/main/java/WebApplication.java

```diff
@@ -85,22 +89,147 @@ public static String render(Map<String, Object> model, String templatePath) {
         // Where does this come from in python?
         model.put("title", "Public timeline");
 
+        var db = new SQLite();
+        var conn = db.getConnection();
+        var statement = conn.prepareStatement(
+                "        select message.*, user.* from message, user\n" +
+                        "        where message.flagged = 0 and message.author_id = user.user_id and (\n" +
+                        "            user.user_id = ? or\n" +
+                        "            user.user_id in (select whom_id from follower\n" +
+                        "                                    where who_id = ?))\n" +
+                        "        order by message.pub_date desc limit ?");
+
+        statement.setInt(1, 1); // TODO: user id
+        statement.setInt(2, 1); // TODO: user id
+        statement.setInt(3, WebApplication.PER_PAGE);
+        ResultSet rs = statement.executeQuery();
+
+        var results = new ArrayList<HashMap<String, Object>>();
+        while (rs.next()) {
+            var result = new HashMap();
+            result.put("message_id", rs.getInt("message_id"));
+            result.put("author_id", rs.getInt("author_id"));
+            result.put("text", rs.getString("text"));
+            result.put("pub_date", rs.getString("pub_date")); // Type?
+            result.put("flagged", rs.getInt("flagged"));
+            result.put("username", rs.getString("username"));
+            result.put("email", rs.getString("email"));
+            result.put("pw_hash", rs.getString("pw_hash"));
+            results.add(result);
+        }
+        model.put("messages", results);
+
         return WebApplication.render(model, WebApplication.Templates.PUBLIC_TIMELINE);
     };
 
+    public static Route serveUserByUsernameTimelinePage = (Request request, Response response) -> {
+        try {
+            Map<String, Object> model = new HashMap<>();
+
+            // TODO: Port flask "flashes"
+
+            model.put("splash", URLS.USER);
+            // Where does this come from in python?
+            model.put("title", "Public timeline");
+
+            var db = new SQLite();
+            var conn = db.getConnection();
+
+            var profileStmt = conn.prepareStatement(
+                    "select * from user where username = ?"
+            );
+            profileStmt.setString(1, request.params(":username"));
+            var profileRs = profileStmt.executeQuery();
+            if (profileRs.isClosed()) {
+                response.status(404);
+                return "404"; // TODO: What does the old one do?
+            }
+            var profileUser = new HashMap<String, Object>();
+            profileUser.put("user_id", profileRs.getInt("user_id"));
+            profileUser.put("username", profileRs.getString("username"));
+            profileUser.put("email", profileRs.getString("email"));
+            model.put("profile_user", profileUser);
+            profileRs.close();
```

Comment:
We might wanna refactor this out into another function, as i expect many of the functions need to return the user from a username.